segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/OperationProcessor.java

```diff
@@ -586,47 +605,33 @@ void failOperation(CompletableOperation operation, Throwable failureCause) {
         }
 
         /**
-         * Rolls back any metadata that is affected by a failure for the given commit args and collects all CompletableOperations
-         * off the pendingOperations queue that are affected. While the metadata is rolled back, the operations themselves
+         * Rolls back any metadata that is affected by a failure for the given commit args and collects all pending
+         * CompletableOperations that are affected. While the metadata is rolled back, the operations themselves
          * are not failed (since this method executes while holding the lock).
          *
-         * @param ex         The Exception that triggered the failure.
          * @param commitArgs The CommitArgs that points to the DataFrame which failed to commit.
-         * @param target     A Map where the failure candidates will be collected. Key=Operation, Value=Failure Exception.
+         * @param candidates A Collection where the failure candidates will be collected.
          */
         @GuardedBy("stateLock")
-        private void collectFailureCandidates(Throwable ex, DataFrameBuilder.CommitArgs commitArgs, Map<CompletableOperation, Throwable> target) {
+        private void collectFailureCandidates(DataFrameBuilder.CommitArgs commitArgs, Collection<CompletableOperation> candidates) {
```

Comment:
Looking at this and the caller I think this would be cleaner if this method returned a collection rather than taking the 'candidates' arg to populate.

Reply:
Good suggestion.